### PR TITLE
Add local variables to notifications

### PR DESCRIPTION
**Pull request by remstone7**
#122

Adds local_variables to request

### CODE REVIEW

**Review by joshuap — COMMENTED** [review]
This is interesting—I didn't know this was possible in Python! Are there any performance implications or anything like that? There are in Ruby, which is one reason why we disable it by default. The other reason is security—the values could be sensitive, and I think reporting local variables is so advanced that it might surprise some people. For that reason I'd suggest we add a new config option similar to the Ruby one, and turn it off by default.

My other suggestion is that we apply the `params_filters` config option to the `local_variables` object so that any sensitive keys are filtered out.

@Kelvin4664 @subzero10 can you review when you get a sec?

**Comment by remstone7** [comment]
@joshuap yup it has it in the `inspect` package here: https://docs.python.org/3/library/inspect.html - this is a built in so no performance impacts, this is similar to what sentry does, the only difference than how i implemented it is that they take the entire traceback, create a frame for each and then get the local variables for each frame (section of the traceback).  other implementations like `inspect.currentframe().f_locals`

Makes sense about the sensitive keys and good call out here, will utilize param filters for that and then will also make it a configurable option 👍 thanks for the feedback!

**Comment by subzero10** [comment]
LGTM. Thank you @remstone7!
I'll wait for the param filtering + config option before approving.

**Comment by remstone7** [comment]
latest push:
using django to test a view but config:
image


inside the app:
image

**Comment by joshuap** [comment]
Looking good here, well let @subzero10 review.

Since it's possible to get local variables for *every* frame of the stack trace, should we plan to modify our notice schema to add a `local_variables` option to the stack frame? We could then implement an expanded local_variables feature in a subsequent version of honeybadger-python.

**Comment by remstone7** [comment]
This would be 🎉 
> Since it's possible to get local variables for _every_ frame of the stack trace, should we plan to modify our notice schema to add a `local_variables` option to the stack frame? We could then implement an expanded local_variables feature in a subsequent version of honeybadger-python.

File: honeybadger/config.py [hunk 1]
```diff
@@ -15,7 +15,8 @@ class Configuration(object):
         ('params_filters', list),
         ('force_report_data', bool),
         ('force_sync', bool),
-        ('excluded_exceptions', list)
+        ('excluded_exceptions', list),
+        ('is_using_local_variables', bool)
```
Comment (subzero10) [inline honeybadger/config.py:19]: How about a `report_local_variables` or `include_local_variables`?

**Review by subzero10 — COMMENTED** [review]
Great! I left a comment but it's not breaking, just a suggestion.

Can we do one last thing? Let's add this new configuration option to the [README.md](https://github.com/honeybadger-io/honeybadger-python#configuration) page.

File: honeybadger/config.py (hunk 1 above)
Comment (remstone7) [reply honeybadger/config.py]: @subzero10, made the changes 👍

**Review by subzero10 — APPROVED** [review]

**Pull request merged**